New package: GALAHAD v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: GALAHAD
- Repository: https://github.com/ralna/GALAHAD
- Created by: @amontoison
- Version: v0.1.0
- Commit: f826fc6f8ea15bd366d80cc33ada9978edd797d9
- Reviewed by: @amontoison
- Reference: https://github.com/ralna/GALAHAD/commit/f826fc6f8ea15bd366d80cc33ada9978edd797d9#commitcomment-138952893
- Description: A library of modern Fortran modules for nonlinear optimization
<!-- 2541789fa6533eb6772c23620d39885a86761d48fc59172050913f816aa5f9a08d991c258e8fc6d668004dc1e2d7572ba86c55869710711d59e081624eaa82062a2cc42a1f55a98d54aa8d2acafd0091c8e1fdfd5575a39b5ff20e771db6eafbee4341a4947ba74b8598c84d2017ff92b7ffcea8452964571b3b9dcca8bc897ce98464ae1e253b3363abdf09977850319202df186aaf0b50ae1428c477cb6c1922c4fac52e5758d45279d492e4470304865d95eb734d872150faad9bd0404a10503c9dd7b893d5ab2d4e7bef3ffb3aa41741d1965ab39cbcd357a0cb3f8631bc -->

amontoison commented:
[noblock]
We would like to keep the name `GALAHAD.jl` to match the Fortran code that it's wrapping.
The associated artifact is `GALAHAD_jll.jl`.
It's similar to the packages `MUMPS.jl`, `HSL.jl`, or `PRIMA.jl`.

GunnarFarneback commented:
[noblock]

I think it's more likely that LICENSE is not detected at all because it's at the top level of the repository and thus outside the Julia package.

amontoison commented:
[noblock]
I should probably add a license in the Julia package (subfolder `GALAHAD.jl`).
Because a GitHub repository can have multiple Julia packages with different license, the registrator (probably) expects a license in each subfolder related to a package.

JuliaRegistrator committed:
New package: GALAHAD v0.1.0

UUID: c591611e-c43e-4808-98fb-ce4cafe3a717
Repo: https://github.com/ralna/GALAHAD.git
Tree: 61f3c384e8fda449adf6a0e0d3bfb5fd3cba511b

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

GunnarFarneback commented:
[noblock]

Those thoughts are all correct but much more importantly only the files inside the package directory get distributed by the package servers and installed on the target computers. Most licences require that the copyright notices etc. are retained when the software is redistributed, so it's really necessary to have a copy of the (or a separate) license within the Julia package.
